Modules/TOF/src/TOFTaskCompressed.cxx

```diff
@@ -0,0 +1,148 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TOFTaskCompressed.cxx
+/// \author Nicolo' Jacazio
+///
+
+// ROOT includes
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH1F.h>
+#include <TH2F.h>
+#include <TH1I.h>
+#include <TH2I.h>
+
+// O2 includes
+#include "DataFormatsTOF/CompressedDataFormat.h"
+#include <Framework/DataRefUtils.h>
+
+// QC includes
+#include "QualityControl/QcInfoLogger.h"
+#include "TOF/TOFTaskCompressed.h"
+
+namespace o2::quality_control_modules::tof
+{
+
+TOFTaskCompressed::TOFTaskCompressed() : TaskInterface(),
+                                         mDecoder(),
+                                         mHits(nullptr),
+                                         mTime(nullptr),
+                                         mTimeBC(nullptr),
+                                         mTOT(nullptr),
+                                         mIndexE(nullptr),
+                                         mSlotEnableMask(nullptr),
+                                         mDiagnostic(nullptr)
+{
+}
+
+TOFTaskCompressed::~TOFTaskCompressed()
+{
+  mHits.reset();
+  mTime.reset();
+  mTimeBC.reset();
+  mTOT.reset();
+  mIndexE.reset();
+  mSlotEnableMask.reset();
+  mDiagnostic.reset();
+}
+
+void TOFTaskCompressed::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info) << "initialize TOFTaskCompressed" << ENDM;
+
+  mHits.reset(new TH1F("hHits", "hHits;Number of hits", 1000, 0., 1000.));
+  getObjectsManager()->startPublishing(mHits.get());
+  mDecoder.mHistos[mHits->GetName()] = mHits.get();
```

Comment:
I am a bit worried about this. You deal with both shared_ptr and raw pointers. What about using shared_ptr inside mHistos ? this way, the internal counting will be correct ? I might be missing somehting though.

Reply:
Yes, you are right, I changed everything to shared pointers now